content/docs/intro/concepts/programming-model.md

```diff
@@ -56,13 +56,26 @@ if err != nil {
 }
 ```
 
+```csharp
+using Pulumi;
+using Pulumi.Aws.Ec2;
+
+// Inside call to Deployment.Run
+var group = new SecurityGroup(...);
+var server = new Instance("webserver-www", new InstanceArgs
+{
+    //...
+    SecurityGroups = { group.Name }, // reference the security group resource above
+});
+```
+
 To publish values that you wish to access outside your application, create a [stack output](#stack-outputs) via module exports.
 
 In Pulumi, you can group multiple resources in a [component](#components). A component is a logical container for physical cloud resources and affects how resources are grouped in the CLI and the pulumi.com console.
 
 ## Programs {#programs}
 
-Pulumi programs are authored in general-purpose programming languages such as [JavaScript]({{< relref "/docs/intro/languages/javascript" >}}) or [Python]({{< relref "/docs/intro/languages/python" >}}). You can use any packages supported by the language's package manager, as well as [Pulumi packages]({{< relref "/docs/reference/pkg" >}}).
+Pulumi programs are authored in general-purpose programming languages such as [JavaScript]({{< relref "/docs/intro/languages/javascript" >}}) or [Python]({{< relref "/docs/intro/languages/python" >}}) or a .NET Language (in preview). You can use any packages supported by the language's package manager, as well as [Pulumi packages]({{< relref "/docs/reference/pkg" >}}).
 
 When `pulumi up` is run, your Pulumi program is run and the Pulumi CLI determines the desired state of application resources. A Pulumi program can reference artifacts that have already been published (such as S3 objects or prebuilt Docker images) or it can define application resources itself so that everything is versioned together. For example, if your program uses `cloud.Service` with a `build` step, or defines a Lambda for an S3 trigger, you're defining application code that is implicitly deployed during the `pulumi up`.
```

Comment:
if this pr adds /docs/intro/languages/dotnet we could have `or a .NET Language (in preview)` link to it.  i coudln't do that in my original change because i wasn't adding that file.